test/integration/standard_filter_test.rb

```diff
@@ -353,9 +353,21 @@ def test_floor
   end
 
   def test_append
-    assigns = {'a' => 'bc', 'b' => 'd' }
-    assert_template_result('bcd',"{{ a | append: 'd'}}",assigns)
-    assert_template_result('bcd',"{{ a | append: b}}",assigns)
+    assigns = {'a' => 'bc', 'b' => 'd', 'c' => [1,2,3,4], 'd' => 10 }
+
+    assert_template_result('bcd'      ,"{{ a | append: 'd'}}" ,assigns)
+    assert_template_result('bcd'      ,"{{ a | append:  b }}" ,assigns)
+    assert_template_result('123410'   ,"{{ c | append:  d }}" ,assigns)
+    assert_template_result('bc1234'   ,"{{ a | append:  c }}" ,assigns)
+    assert_template_result('12341234' ,"{{ c | append:  c }}" ,assigns)
+
+    assert_equal([1,2,3,4],  @filters.append([1,2], [3,4] ))
+    assert_equal([1,2,'a'],  @filters.append([1,2],  'a'  ))
+    assert_equal([1,2, 10],  @filters.append([1,2],  10   ))
+    assert_equal(['a', 1,2], @filters.append('a',  [1,2]  ))
```

Comment:
if I make append always return array, it will change completely their behavior, that was to append strings.  maybe be better have a different filter to append arrays, like I made before.